projects/finance-grid/src/App.tsx

```diff
@@ -0,0 +1,266 @@
+  const intervalRef = useRef<ReturnType<typeof setInterval> | null>(null);
+  useEffect(() => {
+    registerIconFromText("trending_up", TRENDING_UP, "material");
+    registerIconFromText("trending_down", TRENDING_DOWN, "material");
+    dataService.getFinanceData().then((data: any) => {
+      setFinanceData(data);
+    });
+    intervalRef.current = setInterval(() => {
+      dataService.updateAllPrices(gridRef.current!.data);
+      gridRef.current!.markForCheck();
+    }, updateTimerInMs);
+    return () => {
+      if (intervalRef.current) {
+        clearInterval(intervalRef.current);
+      }
+    };
+  }, []);
+
+  const getPathToImage = (val: string): string => {
+    return `${import.meta.env.BASE_URL}companies/${val.split(" ")[0]}.png`;
+  };
+
+  const assetTemplate = (ctx: IgrCellTemplateContext) => {
+    return (
+      <div className="assets-container">
+        <IgrAvatar
+          src={getPathToImage(ctx.cell.value)}
+          shape="rounded"
+        ></IgrAvatar>
+        <span>{ctx.cell.value}</span>
+      </div>
+    );
+  };
+
+  const holdingPeriodTemplate = (ctx: IgrCellTemplateContext) => {
+    return <span>{ctx.cell.value} days</span>;
+  };
+
+  const renderIcon = (val: number) => {
+    return val >= 0 ? (
+      <IgrIcon name="trending_up" collection="material"></IgrIcon>
+    ) : (
+      <IgrIcon name="trending_down" collection="material"></IgrIcon>
+    );
+  };
+
+  const dailyChangePercentageTemplate = (ctx: IgrCellTemplateContext) => {
+    const percentageValue = (ctx.cell.value * 100).toFixed(2);
+    return (
+      <div className="assets-container">
+        <div>{percentageValue}%</div>
+        {renderIcon(ctx.cell.value)}
+      </div>
+    );
+  };
+
+  const profitLossValueTemplate = (ctx: IgrCellTemplateContext) => {
+    const formattedValue = `${ctx.cell.value < 0 ? "-" : ""}$${Math.abs(
+      ctx.cell.value
+    ).toFixed(2)}`;
+    return (
+      <div className="assets-container">
+        <div>{formattedValue}</div>
+        {renderIcon(ctx.cell.value)}
+      </div>
+    );
+  };
+
+  const allocationTemplate = (ctx: IgrCellTemplateContext) => {
+    const percentageValue = (ctx.cell.value * 100).toFixed(2);
+    return (
+      <div className="progress-container">
+        <div>{percentageValue}%</div>
+        <IgrLinearProgress
+          value={ctx.cell.value * 100}
+          animationDuration={0}
+          hideLabel={true}
+        ></IgrLinearProgress>
+      </div>
+    );
+  };
+
+  const profitConditionHandler = (rowData: any, columnKey: string) => {
+    return rowData[columnKey] >= 0;
+  };
+
+  const lossConditionHandler = (rowData: any, columnKey: string) => {
+    return rowData[columnKey] < 0;
+  };
+
+  const profitLossValueClasses = {
+    profitCondition: profitConditionHandler,
+    lossCondition: lossConditionHandler,
+  };
+
+  const filter = (e: any) => {
+    const value = e.target.value;
+    const expressionTree = new IgrFilteringExpressionsTree();
```

Comment:
I think it should work by assigning an object directly and initializing it with its values directly without using `new` keyword and would eliminate the need to `push` in the `filteringOperands` and etc. so its cleaner. Just make the variable has a type of `IgrFilteringExpressionsTree`